common/smoke-tests/shuffle_dependency_order.py

```diff
@@ -0,0 +1,45 @@
+from random import shuffle
+import xml.etree.ElementTree as ET
+import xml.dom.minidom as md
+
+from util import get_xml_root
+
+def scrub_indented_lines(input):
+    return "\n".join([line for line in input.split("\n") if line.strip() != ''])
+
+def pretty_xml(xml_string):
+    parsed = md.parseString(xml_string)
+    return scrub_indented_lines(parsed.toprettyxml(indent='  ', newl="\n"))
+
+def main():
+    root = get_xml_root('./pom.xml')
+    dependencies = root.findall("dependencies/dependency")
+
+    print("==== Original Dependency Order ====")
+    for dependency in dependencies:
+        print(f"\t{dependency.find('artifactId').text}")
+
+    # Shuffle dependencies in place
+    shuffle(dependencies)
```

Comment:
Why are we shuffling the order? Any randomness like this is almost always going to cause reliability issues in the pipeline.

Reply:
I'm pretty sure that in the case of smoke tests we want to rearrange the dependency order as, in theory, that is how we're going to potentially find dependency issues. Being that this is only shuffling the dependencies, this shouldn't cause any issues that aren't product issues. So long as the dependency order is printed out, for reproducibility, I don't see this particular randomization being a problem. Also, with this running in its own pipeline, it's pretty isolated and shouldn't cause issues with other pipelines.

Reply:
Spoke w/ @weshaggard -- deterministic version mismatch searches are going to be better here than nondeterministic randomization.

Such a test would involve resolving the latest GA released packages and then walking the dependencies to find mismatches.

Reply:
Updated bug: https://github.com/Azure/azure-sdk-for-java/issues/7310